enclave-runtime/src/lib.rs

```diff
@@ -402,3 +402,15 @@ fn internal_trigger_parentchain_block_import() -> Result<()> {
 	triggered_import_dispatcher.import_all()?;
 	Ok(())
 }
+
+#[cfg(debug_assertions)]
+#[no_mangle]
+pub extern "C" fn __assert_fail(
+	__assertion: *const u8,
+	__file: *const u8,
+	__line: u32,
+	__function: *const u8,
+) -> ! {
+	use core::intrinsics::abort;
+	abort()
+}
```

Comment:
Can you please give me some context here? Or better, document in the code why this exists? :)

Reply:
Sure thing:
> I also added the __assert_fail function, which was required (see https://github.com/integritee-network/worker/issues/761#issuecomment-1175992810).

The linked issue/comment mentions that `ring` requires this function if compiled in non-release mode. Teaclave added this implementation, but [something changed and newer compiler versions no longer include their implementation](https://github.com/apache/incubator-teaclave-sgx-sdk/issues/373#issuecomment-1035706472). 
> looks like the newer compiler somehow does not include __assert_fail's implementation in generated static libraries. to this end, you can export CFLAGS="-DRING_CORE_NOSTDLIBINC=1" to let Ring build without the __assert_fail dependency.

I though that the `CFLAGS` route is fragile (e.g. something might overwrite it, instead of just appending to it), which is why I opted for adding the implementation.

Do you think it would be better to use `CFLAGS`?

Reply:
No, it is fine, an elaboration in the code suffices for me.

Reply:
Oops, I was very unclear here. I meant to say that this elaboration should be documented in the code, there is no way that anybody guesses why this is here. Sorry for being picky again. :)

Reply:
Sorry, it seems that you already did that, but the diff of the PR is not displayed correctly.

Reply:
Sorry for that, I pushed to the wrong branch :).